api/pod-handlers.go

```diff
@@ -358,46 +371,28 @@ func getDescribePodResponse(session *models.Principal, params operator_api.Descr
 	return retval, nil
 }
 
-func describeStatus(status corev1.ContainerStatus) (*models.State, *models.State) {
+func describeContainerState(status corev1.ContainerState) *models.State {
 	retval := &models.State{}
-	last := &models.State{}
-	state := status.State
-	lastState := status.LastTerminationState
 	switch {
-	case state.Running != nil:
+	case status.Running != nil:
 		retval.State = "Running"
-		retval.Started = state.Running.StartedAt.Time.Format(time.RFC1123Z)
-	case state.Waiting != nil:
+		retval.Started = status.Running.StartedAt.Time.Format(time.RFC1123Z)
+	case status.Waiting != nil:
 		retval.State = "Waiting"
-		retval.Reason = state.Waiting.Reason
-	case state.Terminated != nil:
+		retval.Reason = status.Waiting.Reason
+		retval.Message = status.Waiting.Message
+	case status.Terminated != nil:
 		retval.State = "Terminated"
-		retval.Message = state.Terminated.Message
-		retval.ExitCode = int64(state.Terminated.ExitCode)
-		retval.Signal = int64(state.Terminated.Signal)
-		retval.Started = state.Terminated.StartedAt.Time.Format(time.RFC1123Z)
-		retval.Finished = state.Terminated.FinishedAt.Time.Format(time.RFC1123Z)
-		switch {
-		case lastState.Running != nil:
```

Comment:
Last state whas only being defined when status was Terminated. But this value is being returned regardles of the previous state so we should define it everytime, not only when the current value is terminated.